soh/soh/SaveManager.cpp

```diff
@@ -324,6 +324,8 @@ void SaveManager::Init() {
     const std::filesystem::path sGlobalPath = sSavePath / std::string("global.sav");
     auto sOldSavePath = Ship::Window::GetPathRelativeToAppDirectory("oot_save.sav");
     auto sOldBackupSavePath = Ship::Window::GetPathRelativeToAppDirectory("oot_save.bak");
+    Ship::RegisterHook<Ship::ExitGame>([this]() { ThreadPoolWait(); });
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnExitGame>([this](uint32_t fileNum) { ThreadPoolWait(); });
```

Comment:
do we need both of these?

Reply:
As I understand it, GameInteractor is when quitting/resetting the game to go back to titlescreen/fast file sect, and Ship::Hook is the program exiting the program as a whole. Both would be desirable.

Probably worth adding comments here clarifying both being used

Reply:
Yes, they are two completely different hooks. First one is called on SoH exit (Alt+F4, X button at the top right, etc) while the second seems only to be called when not continuing after a death or when explicitly resetting with Ctrl+R or the menu option. There should probably be a hook inserted into Ship::ExitGame to call GameInteractor::OnExitGame in the future.

Reply:
Sounds like it's just a naming issue. I figure we should either rename `Ship::ExitGame` to `Ship::ExitSoH` or come up with a new name for `GameInteractor::OnExitGame`

I'm not sure if it would make sense to fire off the `GameInteractor::OnExitGame` hook every time we use `Ship::ExitGame`

Reply:
Good call, I like `ExitProgram` or `ExitApplication`

Reply:
@Kenix3 thoughts since this will be an LUS change?